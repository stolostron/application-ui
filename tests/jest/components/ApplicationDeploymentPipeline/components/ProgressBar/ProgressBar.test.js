/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2016, 2019. All Rights Reserved.
 *
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 *******************************************************************************/

const React = require("react");
const renderer = require("react-test-renderer");
const ProgressBar = require("../../../../../../src-web/components/ApplicationDeploymentPipeline/components/ProgressBar")
  .default;

describe("ProgressBar", () => {
  it("ProgressBar renders correctly.", () => {
    const tree = renderer
      .create(<ProgressBar status={[1, 2, 3, 4, 5]} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
