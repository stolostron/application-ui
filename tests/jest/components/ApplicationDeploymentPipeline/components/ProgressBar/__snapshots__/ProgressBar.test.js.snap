// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar ProgressBar renders correctly. 1`] = `
<div
  id="ProgressBar"
>
  <div
    className="percentagesContainers"
  >
    <div
      className="pass"
    >
      <div
        className="percentage"
      >
        40%
      </div>
      <div
        className="resourceType"
      >
        <div
          className="completedSquare"
        />
        <div
          className="textResource"
        >
          Resources completed
        </div>
      </div>
    </div>
    <div
      className="inprogress"
    >
      <div
        className="percentage"
      >
        47%
      </div>
      <div
        className="resourceType"
      >
        <div
          className="inprogressSquare"
        />
        <div
          className="textResource"
        >
          Resources in progress
        </div>
      </div>
    </div>
    <div
      className="fail"
    >
      <div
        className="percentage"
      >
        13%
      </div>
      <div
        className="resourceType"
      >
        <div
          className="failedSquare"
        />
        <div
          className="textResource"
        >
          Resources failed
        </div>
      </div>
    </div>
  </div>
  <div
    className="progressContainer"
    style={
      Object {
        "background": "linear-gradient(to right,  #007D79 0%,#007D79 40%, #92EEEE 40%, #92EEEE 86.66666666666666%, #E0182D 86.66666666666666%, #E0182D 99.99999999999999%, #cccccc 99.99999999999999%, #cccccc 100%)",
      }
    }
  />
</div>
`;
